x-pack/platform/plugins/shared/stack_connectors/server/connector_types/email/send_email_graph_api.ts

```diff
@@ -46,15 +92,47 @@ export async function sendEmailGraphApi(
   }
   const errString = stringify(res.data);
   logger.warn(
-    `error thrown sending Microsoft Exchange email for clientID: ${sendEmailOptions.options.transport.clientId}: ${errString}`
+    `error thrown sending Microsoft Exchange email for clientID: ${options.transport.clientId}: ${errString}`
   );
   throw new Error(errString);
 }
 
-interface SendEmailGraphApiOptions {
-  options: SendEmailOptions;
-  headers: Record<string, string>;
-  messageHTML: string;
+export async function sendEmailWithAttachments(
+  params: SendEmailParams,
+  smallAttachmentLimit: number = SMALL_ATTACHMENT_LIMIT,
+  attachmentChunkSize: number = ATTACHMENT_CHUNK_SIZE
+): Promise<AxiosResponse> {
+  const emailId = await createDraft(params);
+  const attachments = params.sendEmailOptions.attachments;
+  for (const attachment of attachments) {
+    const size = Buffer.byteLength(attachment.content);
+    if (size < smallAttachmentLimit) {
+      // If attachment is smaller than the limit, add the attachment to the draft email
+      await addAttachment(emailId, attachment, params);
```

Comment:
can we add some debug logging for all of this to aid ourselves in the future? like `creating draft email`, `attachment is smaller than 2Mb, attaching to draft`, `attachment is larger than 2 Mb, creating upload session`, `uploadUrl: ${uploadUrl}`, `uploading chunk 1 of ...`, etc.

Reply:
Resolved in this commit, https://github.com/elastic/kibana/pull/220539/commits/894e62fd17aac45bbbcbcca7fc6238f1ae0d4b30